### ISSUE TITLE
/solc is in varying states on each testnet, and the errors are confusing

### ISSUE THREAD

**Issue by jamshidh** [blockapps/bloc#35]
running "bloc" compile on different servers, or turning ssl on or off (with different versions of /solc) yields unpredictable results....  With odd error messages.

http://api.blockapps.net

--------------

> bloc compile
compiling sources
contracts/Payout.sol
contracts/SimpleMultiSig.sol
contracts/SimpleStorage.sol
compile failed with error message: Solidity(code): TypeError: Cannot use 'in' operator to search for 'error' in <html>
<head><title>301 Moved Permanently</title></head>
<body bgcolor="white">
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx/1.4.6 (Ubuntu)</center>
</body>
</html>

compile failed with error message: Solidity(code): TypeError: Cannot use 'in' operator to search for 'error' in <html>
<head><title>301 Moved Permanently</title></head>
<body bgcolor="white">
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx/1.4.6 (Ubuntu)</center>
</body>
</html>

compile failed with error message: Solidity(code): TypeError: Cannot use 'in' operator to search for 'error' in <html>
<head><title>301 Moved Permanently</title></head>
<body bgcolor="white">
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx/1.4.6 (Ubuntu)</center>
</body>
</html>

https://api.
------------------

> bloc compile
compiling sources
contracts/Payout.sol
contracts/SimpleMultiSig.sol
contracts/SimpleStorage.sol
compile failed with error message: Solidity(code): OperationalError: solc: error while loading shared libraries: libboost_filesystem.so.1.54.0: cannot open shared object file: No such file or directory

compile failed with error message: Solidity(code): OperationalError: solc: error while loading shared libraries: libboost_filesystem.so.1.54.0: cannot open shared object file: No such file or directory

compile failed with error message: Solidity(code): OperationalError: solc: error while loading shared libraries: libboost_filesystem.so.1.54.0: cannot open shared object file: No such file or directory

http://testapi.blockapps.net

-------------------

> bloc compile
compiling sources
contracts/Payout.sol
contracts/SimpleMultiSig.sol
contracts/SimpleStorage.sol
compile failed with error message: Solidity(code): Error: HTTPRequest failed:
Error: connect ECONNREFUSED 45.55.142.6:80
compile failed with error message: Solidity(code): Error: HTTPRequest failed:
Error: connect ECONNREFUSED 45.55.142.6:80
compile failed with error message: Solidity(code): Error: HTTPRequest failed:
Error: connect ECONNREFUSED 45.55.142.6:80

https://testapi.blockapps.net

-------------

> bloc compile
compiling sources
contracts/Payout.sol
contracts/SimpleMultiSig.sol
contracts/SimpleStorage.sol
compile failed with error message: Solidity(code): TypeError: Cannot use 'in' operator to search for 'error' in Internal Error:
solc: readCreateProcessWithExitCode: runInteractiveProcess: exec: does not exist (No such file or directory)

compile failed with error message: Solidity(code): TypeError: Cannot use 'in' operator to search for 'error' in Internal Error:
solc: readCreateProcessWithExitCode: runInteractiveProcess: exec: does not exist (No such file or directory)

compile failed with error message: Solidity(code): TypeError: Cannot use 'in' operator to search for 'error' in Internal Error:
solc: readCreateProcessWithExitCode: runInteractiveProcess: exec: does not exist (No such file or directory)

http://hacknet.blockapps.net

-------------------

> bloc compile
compiling sources
contracts/Payout.sol
contracts/SimpleMultiSig.sol
contracts/SimpleStorage.sol
compile successful, writing meta/Payout.json
compile successful, writing meta/SimpleMultiSig.json
compile successful, writing meta/SimpleStorage.json

https://hacknet.blockapps.net

------------------

> bloc compile
compiling sources
contracts/Payout.sol
contracts/SimpleMultiSig.sol
contracts/SimpleStorage.sol
compile successful, writing meta/Payout.json
compile successful, writing meta/SimpleStorage.json
compile successful, writing meta/SimpleMultiSig.json

**Comment by ryanreich** [comment]
The 301 responses are the nginx forwarding from http to https.  The
TypeError responses are an indication that solc couldn't be run (say,
wasn't in the path).  The shared library errors mean what they say,
probably a sign of copying a precompiled solc onto the wrong platform.  I
do not know what's up with testapi, though: I've never seen ECONNREFUSED
before.

I think all of them can be fixed simply by installing my binary solc
distribution (which has all the necessary dynamic libraries).  You can grab
the `~/solc` directory and the `~/.local/bin/solc` script (which has to be
modified b/c of the different username) from any Azure deployment and copy
them directly anywhere.  The script doesn't have to be in `.local/bin`, but
it does have to be in PATH; ~/solc` should remain at that exact path and
*not* be in `PATH`.

**Comment by kjameslubin** [comment]
Is this resolved @ryanreich @jamshidh?

**Comment by kejace** [comment]
We also want `bloc` to report on the frontend the `solc` version that is used.

**Comment by ryanreich** [comment]
Not at all, sadly. When the p2p stuff has proceeded I can finally upgrade
solc to the newest version.
